.github/actions/generate-coverage/tests/test_scripts.py

```
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import subprocess
+from pathlib import Path
+
+
+def run_script(script: Path, env: dict[str, str], *args: str) -> subprocess.CompletedProcess:
+    cmd = ["uv", "run", "--script", str(script), *args]
+    return subprocess.run(cmd, capture_output=True, text=True, env=env)
```

Ruff (0.11.9)

`subprocess` call: check for execution of untrusted input

(S603)